source/fundamentals/connection/connection-options.txt

```diff
@@ -0,0 +1,225 @@
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 1 1
+
+   * - Name
+     - Accepted Values
+     - Default Value
+     - Description
+
+   * - **appname**
+     - string
+     - ``null``
+     - The app name the driver passes to the server in the client
+       metadata as part of the connection handshake. The server prints this value to the MongoDB logs once the connection is established. The value is also recorded in the slow query logs and profile collections.
+
+   * - **authMechanism**
+     - string
+     - ``null``
+     - The authentication mechanism the driver uses when connecting to the
+       server. If you don't include this option, the driver uses either ``SCRAM-SHA-1`` or ``SCRAM-SHA-256``, depending on the server version. 
+       
+       See
+       :ref:`authentication mechanisms <csharp-authentication-mechanisms>` for available
+       authentication mechanisms.
+
+   * - **authMechanismProperties**
+     - comma-separated ``key:value`` pairs (for example, ``opt1:val1,opt2:val2``)
+     - ``null``
+     - Other authentication options, such as enabling hostname canonicalization for  GSSAPI.
+
+   * - **authSource**
+     - string
+     - ``null``
+     - The database that connections should authenticate against.
+
+   * - **compressors**
+     - comma-separated list of strings (for example, ``snappy,zlib,zstd``)
+     - ``null``
+     - The preferred compression types, in order, for wire-protocol messages
+       sent to or received from the server. The driver uses the first of these compression types that the server supports.
+       
+       See :ref:`csharp-network-compression` for more information.
+
+   * - **connectTimeoutMS**
+     - nonnegative integer
+     - ``30000``
+     - The length of time the driver tries to establish a single TCP
+       socket connection to the server before timing out. Specifying ``0`` disables the connection timeout.
+
+   * - **directConnection**
+     - Boolean
+     - ``false``
+     - Specifies whether to force dispatch **all** operations to the host
+       specified in the connection URI.
+
+   * - **heartbeatFrequencyMS**
+     - integer greater than or equal to 500
+     - ``10000``
+     - The interval between regular server-monitoring checks.
+
+   * - **journal**
+     - Boolean
+     - ``null``
+     - Specifies the default write concern ``"j"`` field for the client. See
+       :ref:`write concern <wc-j>` for more information.
+
+   * - **loadBalanced**
+     - Boolean
+     - ``false``
+     - Specifies whether the driver is connecting to a load balancer.
+
+   * - **localThresholdMS**
+     - nonnegative integer
+     - ``15``
+     - The latency window for server eligibility. If a server's round trip takes longer than the fastest server's round-trip time plus this value, the server isn't eligible for selection. Specifying ``0`` selects the fastest eligible server.
+
+   * - **maxIdleTimeMS**
+     - nonnegative integer
```

Comment:
issue: I wonder if it would be more idiomatic to say "unsigned" since C# has that data type.

Reply:
Interesting question. An unsigned integer _is_ more idiomatic to C#; on the other hand:

- This page describes options in the context of the connection string, where there's no such thing as a data type.
- "Nonnegative integer" stays in sync with the spec.
- "Unsigned integer" is less intuitive, IMO.

What do you think?

Reply:
I'll buy it as long as we have reasoned it out.